src/software/ai/hl/stp/play/free_kick/free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,245 @@
+
+    updateAlignToBallTactic(event.common.world_ptr);
+    tactics_to_run[0].emplace_back(align_to_ball_tactic);
+
+    updateOffensivePositioningTactics(event.common.world_ptr);
+    tactics_to_run[0].emplace_back(offensive_positioning_tactics[0]);
+    tactics_to_run[0].emplace_back(offensive_positioning_tactics[1]);
+
+    crease_defender_tactics[0]->updateControlParams(
+        event.common.world_ptr->ball().position(),
+        TbotsProto::CreaseDefenderAlignment::LEFT);
+    crease_defender_tactics[1]->updateControlParams(
+        event.common.world_ptr->ball().position(),
+        TbotsProto::CreaseDefenderAlignment::RIGHT);
+    tactics_to_run[0].emplace_back(crease_defender_tactics[0]);
+    tactics_to_run[0].emplace_back(crease_defender_tactics[1]);
+
+    event.common.set_tactics(tactics_to_run);
+}
+
+bool FreeKickPlayFSM::setupDone(const Update &event)
+{
+    if (align_to_ball_tactic->done())
+    {
+        LOG(DEBUG) << "Finished aligning to ball.";
+    }
+    return align_to_ball_tactic->done();
+}
+
+void FreeKickPlayFSM::updateOffensivePositioningTactics(const WorldPtr &world_ptr)
+{
+    using Zones = std::unordered_set<EighteenZoneId>;
+
+    auto pass_eval = pass_generator.generatePassEvaluation(*world_ptr);
+
+    for (unsigned int i = 0; i < offensive_positioning_tactics.size(); i++)
+    {
+        Zones zone = {ranked_zones[i]};
+        auto pass  = pass_eval.getBestPassInZones(zone).pass;
+
+        offensive_positioning_tactics[i]->updateControlParams(
+            pass.receiverPoint(), pass.receiverOrientation(), 0.0);
+    }
+}
+
+void FreeKickPlayFSM::updateAlignToBallTactic(const WorldPtr &world_ptr)
+{
+    // We want the kicker to get into position behind the ball facing the enemy net
+    Point ball_pos                = world_ptr->ball().position();
+    Vector ball_to_enemy_goal_vec = world_ptr->field().enemyGoalCenter() - ball_pos;
+    align_to_ball_tactic->updateControlParams(
+        ball_pos - ball_to_enemy_goal_vec.normalize(ROBOT_MAX_RADIUS_METERS * 2),
+        ball_to_enemy_goal_vec.orientation(), 0);
+}
+
+bool FreeKickPlayFSM::shotFound(const Update &event)
+{
+    shot = calcBestShotOnGoal(event.common.world_ptr->field(),
+                              event.common.world_ptr->friendlyTeam(),
+                              event.common.world_ptr->enemyTeam(),
+                              event.common.world_ptr->ball().position(), TeamType::ENEMY);
+    return shot &&
+           shot->getOpenAngle() >
+               Angle::fromDegrees(
+                   ai_config.attacker_tactic_config().min_open_angle_for_shot_deg());
+}
+
+void FreeKickPlayFSM::shootBall(const Update &event)
+{
+    LOG(DEBUG) << "Shooting ball...";
+    PriorityTacticVector tactics_to_run = {{}};
+
+    Point ball_pos = event.common.world_ptr->ball().position();
+
+    shoot_tactic->updateControlParams(
+        ball_pos, (shot->getPointToShootAt() - ball_pos).orientation(),
+        BALL_MAX_SPEED_METERS_PER_SECOND);
```

Comment:
We shouldn't kick at max speed since it's easy to get shots disallowed. In #3200 I've added a new constant for this (https://github.com/UBC-Thunderbots/Software/pull/3200/files#diff-bea38a7082c851b7c2f575db4b775334bb57f58d767a07617eb39b31ddf22d35), but for now you could probably just do the following
```suggestion
    shoot_tactic->updateControlParams(
        ball_pos, (shot->getPointToShootAt() - ball_pos).orientation(),
        BALL_MAX_SPEED_METERS_PER_SECOND - 0.5);
```